posts/2024-07-25-Rust-1.80.0.md

```diff
@@ -0,0 +1,183 @@
+---
+layout: post
+title: "Announcing Rust 1.80.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.80.0. Rust is a programming language empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via `rustup`, you can get 1.80.0 with:
+
+```console
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.80.0](https://doc.rust-lang.org/nightly/releases.html#version-1800-2024-07-25).
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.80.0 stable
+
+### `LazyCell` and `LazyLock`
+
+These "lazy" types delay the initialization of their data until first access. They are similar to the `OnceCell` and `OnceLock` types [stabilized in 1.70](https://blog.rust-lang.org/2023/06/01/Rust-1.70.0.html#oncecell-and-oncelock), but with the initialization function included in the type.
+
+`LazyLock` is the thread-safe option, making it suitable for places like `static` values. For example, both the `spawn` thread and the main `scope` will see the exact same duration below, since `LAZY_TIME` will be initialized once, by whichever ends up accessing the static first. Neither use has to know *how* to initialize it though, like you would with `OnceLock::get_or_init`.
+
+```rust
+use std::sync::LazyLock;
+use std::time::Instant;
+
+static LAZY_TIME: LazyLock<Instant> = LazyLock::new(Instant::now);
+
+fn main() {
+    let start = Instant::now();
+    std::thread::scope(|s| {
+        s.spawn(|| {
+            println!("Thread lazy time is {:?}", LAZY_TIME.duration_since(start));
+        });
+        println!("Main lazy time is {:?}", LAZY_TIME.duration_since(start));
+    });
+}
+```
+
+`LazyCell` does the same thing without thread synchronization, so it doesn't implement `Sync` needed for `static`, but it can still be used in `thread_local!` statics (with distinct initialization per thread). Either type can also be used in other data structures as well, depending on thread-safety needs, so lazy initialization is available everywhere!
```

Comment:
```suggestion
`LazyCell` does the same thing without thread synchronization, so it doesn't implement `Sync`, which is needed for `static`, but it can still be used in `thread_local!` statics (with distinct initialization per thread). Either type can also be used in other data structures as well, depending on thread-safety needs, so lazy initialization is available everywhere!
```